Improve Text Formatting for Summary and AI Perspective, Fix Export Logic, Enhance UI, and Resolve Build Errors

SharkyBytes opened the pull request:
->Formatted the summary and AI perspective to improve readability and remove LaTeX artifacts.
->Applied React Markdown for better text presentation.
->Disabled the Save button until both the summary and perspective are generated.
->Fixed the Export to PDF and DOC logic, replacing hardcoded data with dynamically formatted content.
->Removed LaTeX commands from the exported files to ensure properly formatted output.
->Fixed the Build Errors and the Hydration errors from the Previous PR

@ManavSarkar Please have a look

Disabling Save Button
![image](https://github.com/user-attachments/assets/89cad748-0ee9-4ed8-a391-351b275b3cc4)

After Changes : Looking like this 

![image](https://github.com/user-attachments/assets/88d35605-ebf1-4d9f-ad55-e3738f6d8d7b)
![image](https://github.com/user-attachments/assets/d3d4878b-46cf-438a-a2d5-018d32558d39)

After exporting into PDF
![image](https://github.com/user-attachments/assets/c146ef40-6672-494c-98d1-42cd84f99028)

After exporting into TXT file
![image](https://github.com/user-attachments/assets/791c6ec1-a4c1-4b6d-99a0-cf93090a67a1)

SharkyBytes commented:
Hey @sagarkawad, I just reviewed your changes— I appreciate your efforts. However, I noticed that the response was initially coming with LaTeX formatting, which was then stringified into JSON. After your update, I checked and found that ReactMarkdown occasionally encountered rendering issues—about 1 in 5 times—due to text overflow and the presence of \latex and \boxed text. These elements made it difficult for ReactMarkdown to process correctly.

To address this, I removed those elements so that now it process the React Markdown thing after cleaning and refined the export functionality, which was previously using hardcoded data. Now, both Export to DOC and Export to PDF work seamlessly with properly formatted content.

One small suggestion: When naming PRs, it’s helpful to include all the key changes you’ve made. I noticed that some of the work you did wasn’t reflected in the title, which might make it harder for reviewers to quickly grasp the full scope of the update. No worries for now—I’ve adjusted it accordingly. Also, no issues with your PR, as when it got merge, your some changes will still exist as you worked on another thing also in the PR. Let's keep this for now. After the merging if you face any error we can work on this issue together! Would love to work.

Here’s how it looks after my changes:

![image](https://github.com/user-attachments/assets/44bb46a9-3c83-4f83-907e-0cf192fb65cc)

![image](https://github.com/user-attachments/assets/499302c4-609a-43dc-a9fe-eee9ca07787f)

sagarkawad commented:
Hi @SharkyBytes what I am talking about is that LaTeX artifacts and ReactMarkdown are two different things. 

Good to see that you fixed the LateX issue before I could. But I am specifically talking about the ReactMarkdown here as you've mentioned in you PR (->Applied React Markdown for better text presentation). 
The ReactMarkDown was still working on my PR #49. 

Just by fixing the LaTeX without interfering with the ReactMarkdown,
Here's the graphical representation - 
<img width="907" alt="Screenshot 2025-03-14 at 11 54 48 AM" src="https://github.com/user-attachments/assets/5b83a741-0922-4324-967d-00e262b784c0" />

What if I create a new PR including fixing the LaTeX issue. I just think that is not how it should be. Right?

SharkyBytes commented:
Hey @sagarkawad! 
I am not saying you to close your PR man! Our work is truly appreciated. I acknowledge that I may have missed your PR initially due to not having proper title, and even I appreciate the effort you put into it. However, there were some issues in the PR that needed to be addressed, which I did with my major work. I worked on resolving the errors, including build issues and incorrect rendering. Additionally, I ensured that the export functionality was properly integrated, as the previous implementation focused primarily on the UI but missed some functional aspects. My intention was never to override anyone's work but rather to refine and enhance it for the project's overall improvement. And still No problem Sagar, let me force push your markdown code only with my latex changes. I guess that would be a fair deal for you.

I believe open source is about collaboration and making the project better for everyone. My goal with Perspective is to ensure it becomes a well-functioning project that people can use, and I take pride in contributing to that. If you still feel that I should revert my PR, I’m open to discussing it. My priority is to make Perspective a valuable and successful project for all contributors.

Looking forward to your thoughts @sagarkawad.

sagarkawad commented:
Yeah sure @SharkyBytes  
Kudos to you for the work done!

Jk-karunadu commented:
@SharkyBytes this fixes which issue ?

Jk-karunadu commented:
@SharkyBytes we all follow the rule that , first create issue then start working on it , like this work makes duplicate of work , try to understand

SharkyBytes commented:
Hey @sagarkawad, I just added this because after fixing it, everything is rendering properly in Markdown. But yeah, no problem, @sagarkawad. I’ve updated the description—check it out!
